plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestS3WithDoubleSlashPathAndS3ASchemeTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.parallel.Execution;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import static io.trino.plugin.iceberg.IcebergQueryRunner.ICEBERG_CATALOG;
+import static io.trino.plugin.iceberg.IcebergQueryRunner.ICEBERG_CATALOG2;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+@Testcontainers
+public class TestS3WithDoubleSlashPathAndS3ASchemeTest
+        extends AbstractTestQueryFramework
+{
+    @Container
+    private static final LocalStackContainer LOCALSTACK = new LocalStackContainer(DockerImageName.parse("localstack/localstack:3.3.0"))
+            .withServices(LocalStackContainer.Service.S3);
+
+    private final String bucketName = "test-bucket";
+
+    private S3Client s3;
+
+    @AfterAll
+    public void tearDown()
+    {
+        if (s3 != null) {
+            s3.close();
+            s3 = null;
+        }
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        s3 = S3Client.builder()
+                .endpointOverride(LOCALSTACK.getEndpointOverride(LocalStackContainer.Service.S3))
+                .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("accesskey", "secretkey")))
+                .region(Region.of(LOCALSTACK.getRegion())).build();
+        s3.createBucket(builder -> builder.bucket(bucketName));
+        return IcebergQueryRunner.builder()
```

Comment:
```
        QueryRunner queryRunner = DistributedQueryRunner.builder(session)
                .build();
         // taking from IcebergQueryRunner only what is necessary for this test
Path dataDir = metastoreDirectory.map(File::toPath).orElseGet(() -> queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data"));
                queryRunner.installPlugin(new TestingIcebergPlugin(dataDir));
                queryRunner.createCatalog(ICEBERG_CATALOG, "iceberg", icebergProperties.buildOrThrow());
                schemaInitializer.orElseGet(() -> SchemaInitializer.builder().build()).accept(queryRunner);

                if (catalog2IcebergProperties.isPresent()) {
                    queryRunner.createCatalog(ICEBERG_CATALOG2, "iceberg", catalog2IcebergProperties.get().buildOrThrow());
                    catalog2SchemaInitializer.orElseGet(() -> SchemaInitializer.builder().build()).accept(queryRunner);
                }
```

Reply:
Use meanignful catalog names: if you go down this path, then you can name the catalogs as "iceberghadoops3" & "icebergnatives3" instead of "iceberg", "iceberg2"

Reply:
Yeah good idea, added test covering canonical and non-canonical paths with different schemes.

Reply:
Agreed, created a queryrunner in the test by adding two catalogs. It looks better and doesn't require to change IcebergQueryRunner
Changed the catalog names as well.